flink-runtime/src/main/java/org/apache/flink/runtime/client/JobExecutionException.java

```diff
@@ -19,41 +19,67 @@
 package org.apache.flink.runtime.client;
 
 import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.clusterframework.ApplicationStatus;
 import org.apache.flink.util.FlinkException;
 
 /**
  * This exception is the base exception for all exceptions that denote any failure during
- * the execution of a job.
+ * the execution of a job, and signal the failure of an application with a given
+ * {@link ApplicationStatus}.
  */
 public class JobExecutionException extends FlinkException {
 
 	private static final long serialVersionUID = 2818087325120827525L;
 
 	private final JobID jobID;
 
+	private final ApplicationStatus status;
+
 	/**
 	 * Constructs a new job execution exception.
 	 *
-	 * @param jobID The job's ID.
-	 * @param msg The cause for the execution exception.
-	 * @param cause The cause of the exception
+	 * @param jobID  The job's ID.
+	 * @param status The application status.
+	 * @param msg    The cause for the execution exception.
+	 * @param cause  The cause of the exception
 	 */
-	public JobExecutionException(JobID jobID, String msg, Throwable cause) {
+	public JobExecutionException(JobID jobID, ApplicationStatus status, String msg, Throwable cause) {
 		super(msg, cause);
 		this.jobID = jobID;
+		this.status = status;
```

Comment:
Shouldn't we assert that `status` is not `SUCCEEDED`?

Reply:
@tillrohrmann , no, there is a case that `jobResult.toJobExecutionResult` throws excepton with SUCCESS application status.

Reply:
Which case is this exactly? I thought that we would obtain a `JobExecutionResult` if the execution of the job was successful.

Reply:
@tillrohrmann , the case is that the execution of the job is successful, but calls `AccumulatorHelper.deserializeAccumulators` throws Exception while calling `jobResult.toJobExecutionResult`.

Reply:
Wouldn't we then fail with an `IOException` being thrown from `JobResult#toJobExecutionResult`?

Reply:
In `MinCluster#executeJobBlocking`, if `JobResult#toJobExecutionResult` throws the `IOException`, the method would catch the `IOException` and throws the `JobExecutionException`. Meanwhile, the application status of `JobExecutionException` is `SUCCEEDED `. Right?